source/common/common/key_value_store_base.cc

```diff
@@ -51,14 +60,28 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
   while (!contents.empty()) {
     absl::optional<absl::string_view> key = getToken(contents, error);
     absl::optional<absl::string_view> value;
+    absl::optional<std::chrono::seconds> ttl = absl::nullopt;
     if (key.has_value()) {
       value = getToken(contents, error);
     }
     if (!key.has_value() || !value.has_value()) {
       ENVOY_LOG(warn, error);
       return false;
     }
-    addOrUpdate(key.value(), value.value(), absl::nullopt);
+    if (checkForTtl(contents)) {
+      uint64_t ttlInt;
+      auto token = getToken(contents, error);
+      if (token.has_value()) {
+        if (absl::SimpleAtoi(token.value(), &ttlInt)) {
+          ttl.emplace(std::chrono::duration_cast<std::chrono::seconds>(
+              std::chrono::time_point<std::chrono::steady_clock>(std::chrono::seconds(ttlInt)) -
+              time_source_.monotonicTime()));
+          std::cout << ttl.value().count();
+        }
+      }
```

Comment:
may want to at least log that we got a malformed TTL value (could indicate a problem in parsing)